CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+# Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.13.4)
+
+project(Chapel VERSION 1.29.0)
+
+# This built-in module allows us to print variables easily, using
+# cmake_print_variables(MY_VARIABLE)
+include(CMakePrintHelpers)
+
+# option(CHPL_LLVM_COMP_ARGS "Override compile args for LLVM")
+# option(CHPL_LLVM_LINK_ARGS "Override linker args for LLVM")
+set(CHPL_CMAKE_HOME CACHE STRING "Override home directory for Chapel")
+
+if (NOT DEFINED ${CHPL_CMAKE_HOME})
+  message(STATUS "CHPL_CMAKE_HOME not set, using $ENV{CHPL_HOME}")
+  set(CHPL_CMAKE_HOME $ENV{CHPL_HOME} CACHE STRING "Override home directory for Chapel" FORCE)
+endif()
+
+# Options that correspond to printchplenv --internal --compiler
+set(CHPL_CMAKE_HOST_PLATFORM CACHE STRING "Description")
```

Comment:
IMO these should just be e.g. `CHPL_HOST_PLATFORM` but there may be a reason I'm not thinking of to separate them.

I might be wrong, but I thought that, for the Makefiles, we needed to separate them because `make` has some automatic behavior where environment variables override Makefile variables.

Reply:
I just added it to be like the Makefile names. They could be matched to the `CHPL_` names though.

Reply:
OK. I have a pretty strong preference for trying to make them match, if we can get that to work.